cartography/intel/rapid7/util.py

```diff
@@ -0,0 +1,246 @@
+            )
+
+    logger.warning("extract_rapid7_configurations_subscriptionid: %s", subscription_id)
+    return subscription_id
+
+
+def extract_rapid7_configurations_resourcegroup(configurations: Any[str, list]) -> str:
+    """
+    Extract from json, specific keyvalue pair: azure resource group
+
+    Input: dataframe row configurations column
+    """
+
+    if configurations is None or not isinstance(configurations, list):
+        return ""
+
+    resource_group = ""
+    for line in configurations:
+        if line["name"] == "azure":
+            logger.debug("line: %s", line)
+            if isinstance(line["value"], str):
+                azure = json.loads(line["value"])
+            else:
+                azure = line["value"]
+            resource_group = extract_rg_from_resourceid(azure["resourceId"])
+
+    logger.warning("extract_rapid7_configurations_resourcegroup: %s", resource_group)
+    return resource_group
+
+
+def extract_rapid7_configurations_resourceid(configurations: Any[str, list]) -> str:
+    """
+    Extract from json, specific keyvalue pair: azure resource id
+
+    Input: dataframe row configurations column
+    """
+
+    if configurations is None or not isinstance(configurations, list):
+        return ""
+
+    resource_id = ""
+    for line in configurations:
+        if line["name"] == "azure":
+            logger.warning("line: %s", line)
+            if isinstance(line["value"], str):
+                azure = json.loads(line["value"])
+            else:
+                azure = line["value"]
+            resource_id = azure["resourceId"]
+
+    logger.warning("extract_rapid7_configurations_resourceid: %s", resource_id)
+    return resource_id
+
+
+def extract_rapid7_configurations_instanceid(configurations: Any[str, list]) -> str:
+    """
+    Extract from json, specific keyvalue pair: azure instance id
+
+    Input: dataframe row configurations column
+    """
+
+    if configurations is None or not isinstance(configurations, list):
+        return ""
+
+    instance_id = ""
+    for line in configurations:
+        if line["name"] == "azure":
+            logger.warning("line: %s", line)
+            if isinstance(line["value"], str):
+                azure = json.loads(line["value"])
+            else:
+                azure = line["value"]
+            instance_id = azure["instanceId"]
+
+    logger.warning("extract_rapid7_configurations_instanceid: %s", instance_id)
+    return instance_id
+
+
+def rapid7_hosts(
+    authorization: Tuple[str, str, str, bool],
+    limit: int = 10000,
+    page: int = 0,
+    sort: str = "",
+    nexpose_timeout: int = 5,
+) -> List:
+    """
+    Get Nexpose inventory full data to array
+    https://help.rapid7.com/insightvm/en-us/api/index.html#operation/getAssets
+
+    To filter, use https://help.rapid7.com/insightvm/en-us/api/index.html#operation/findAssets
+
+    Warning! if very large assets count (>10k, likely depending on assets details), it
```

Comment:
How much memory in megabytes could this be?

Reply:
Even w 48GB RAM, only up to 13k...
Added downloadreport api or local report options to avoid this

Reply:
Do we still need this in the docstring?